SyllabusPlusSchedulerService/SchedulerService.cs

```diff
@@ -125,27 +147,62 @@ private void ScheduleRecordings()
                                             this.configSettings.PanoptoSite,
                                             this.configSettings.PanoptoUserName,
                                             this.configSettings.PanoptoPassword))
+                                    using (SessionManagementWrapper sessionManagementWrapper
+                                        = new SessionManagementWrapper(
+                                        this.configSettings.PanoptoSite,
+                                        this.configSettings.PanoptoUserName,
+                                        this.configSettings.PanoptoPassword))
                                     {
                                         // Schedule session id will determine if need to create or update/delete the corresponding schedule
                                         if (schedule.ScheduledSessionId == null || schedule.ScheduledSessionId == Guid.Empty)
                                         {
                                             log.Debug(schedule.SessionName + " is not associated with a session on the Panopto database. Attempting to schedule.");
-                                            result = remoteRecorderManagementWrapper.ScheduleRecording(schedule);
+                                            result = VerifyFolderAndScheduleRecording(sessionManagementWrapper, schedule, remoteRecorderManagementWrapper);
                                         }
                                         else
                                         {
                                             log.Debug(schedule.SessionName + " is already associated with a session on the Panopto database. Attempting to update schedule.");
-                                            result = remoteRecorderManagementWrapper.UpdateRecordingTime(schedule);
+                                            Session scheduledSession = sessionManagementWrapper.GetSessionById((Guid)schedule.ScheduledSessionId);
+
+                                            // Check if either the primary or secondary remote recorder changed. If so, delete the old session and create a new session.
+                                            if (schedule.PrimaryRemoteRecorderId != scheduledSession.RemoteRecorderIds[0]
```

Comment:
There always is a primary. Secondary could be null.

The second part of the expression is an && expression so the first part will be evaluated and if it is null, it will short circuit. I mainly tested with just a primary recorder to verify this behavior but I did test with a secondary recorder a few times as well.